### PR TITLE
Cultural Naming Improvements (Elven Group)

### PR DESCRIPTION
**Pull request by Loralius**
## Changelog:
- Significantly increased library character and dynastic names for High Elves, Blood Elves, Night Elves, Highborne, Nightborne, Satyr and Naga using NPC names and lore friendly derived naming conventions as a guide.
  - Removed generated first names for Blood Elves and High Elves. 
  - Reduced generated first names for Night Elves and Highborne by one half.
  - Reduced generated dynasty names for Night Elves by half.
  - Replaced generated Highborne dynasty names with selected Kaldorei placeholder dynasty names.
  - Introduced some cross-cultural dynastic names where appropriate.

## Developer changelog:

- Added WoW/RTS NPC names to High Elves, Blood Elves, Night Elves, Nighborne and Naga.
- Dynastic names with `"dynn_x"` point to historical dynasties described in `common/dynasties`. Additions without `"dynn_x"` are lore sourced but not described in 'common/dynasties`.
- Updated formatting in-line with #820 ; categories ordered by lore, lore-inspired (including from other subcultures), random generated.
- Pointed blood elf "dynn_x" defined in common\dynasties to blood elf culture and removed incorrect "dynn_" prefixes for high elven dynasties.
- Added missing dynn_Hawkspear, Brightsun, Salonar from bottom of Commonn\Dynasties to Blood Elf culture
- Replaced duplicate random dynasty names and thematically ineffective random dynasty names with lore names 
  - (Changelog incomplete; check changed files)

<details>
<summary> Click to show </summary>

Blackwing -> Dawnbreeze
Rosestrider -> Brightsong
Firebender -> Snowdawn
Phoenixwalker -> Daybreak
Bloodwood -> Bloodwrath
Dayreaver -> Daybreaker
Dayspear -> Dayblaze
Emberseeker -> Emberfall
Silvergaze -> Silversun
Stargaze -> Starsorrow

Flameflame -> Flameheart
Dewblade -> Dawnfury
Dewbringer -> Dawnstrike
Dewspear -> Dawnglaive
Heartheart -> Heartshade
Redseeker -> Redmorn (Blood Elf)
Redseeker -> Rosebreeze (High Elf)
Blackspell -> Brighthand (High Elf)
Blackspell -> Bloodblade (Blood Elf)
Blackfeather -> Bloodfire
Darkfury -> Dawnshield
Heartsprinter -> Morningsong
Coldbringer -> Ebonscorch
Coldheart -> Emberbinder
Coldshadow -> Brightstar
Downfall -> Sunhollow
Redbringer -> Riverwind (Blood Elf)
Redbringer -> Riverwind (High Elf)
Redforge -> Solarforge
Phoenixgaze -> Sunborn
Phoenixdown -> Truehair
Nightsworn -> Nightscorn
Colddown -> Cinderweave
Downflare -> Darkflare
Mirthsprinter -> Morrowsong
Mirthbinder -> Magehand
Downblade -> Everspring
Blackforge -> Lighthammer
Dewspell -> Dewspring
Downgaze -> Dawnrunner
Of the Nightfall -> Netherburn
Mirthflare -> Mourningsun
Darkdown -> Darkleaf
Autumnshade -> Springrunner
Coldwood -> Autumnwood
Morningbringer -> Morningstar
Moonbringer -> Moonvale (Blood Elf)
Moonbringer -> Loresong (High Elf)
Leafdown -> Feltouch
Leafblossom -> Dawnrose
Lightwood -> Lightblade
Heartwhisper -> Heartspear
Redbrook -> Roseblade
Leaflight -> Lightfall
Coldsworn -> Sunshadow
Nightsky -> Sunlance (High Elf)
Nightsky -> Azuresong (Blood Elf)
Morningshadow -> Sunshroud
Nightvale -> Belore

</details>

## Tests:
- [x] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
   - For all newly added first names, this error is returned:
- [x] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:
- Check if character and dynasty names from lore and lore-inspired categories appear when generating cultural names in the ruler designer. 
- Ensure no unintended text such as slashes or prefixes like "dynn_x" show up. 
- Review added character and dynasty names for suggestions especially regarding enduring character/dynasty names across subcultures.

### CODE REVIEW

File: common/culture/cultures/elf.txt [hunk 1]
```diff
@@ -20,240 +20,370 @@
...
+			"dynn_Brightstar"
+			"dynn_Brightwing"
+			"dynn_Dawnbreeze"
+			"dynn_Dawnrunner"
+			"dynn_Dawnseeker" 
+			"dynn_Dawnshield"
+			"dynn_Daybreak" 
+			"dynn_Ebonscorch"
+			"dynn_Everspring"
+			"dynn_Magehand"
```
Comment (zumbak04) [inline common/culture/cultures/elf.txt:34]: Keep a few names on the same line.
Otherwise, it spams the file.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: localization/english/dynasties/wc_dynasty_names_l_english.yml [hunk 2]
```diff
@@ -2458,8 +2457,271 @@
  ### Kobold ###
 
  ### High Elf ###
+ dynn_Brightsong:0 "Brightsong"
+ dynn_Brightstar:0 "Brightstar"
+ dynn_Dawnbreeze:0 "Dawnbreeze"
```
Comment (zumbak04) [inline localization/english/dynasties/wc_dynasty_names_l_english.yml:2462]: New localization lines always should be at the end.

File: localization/english/dynasties/wc_dynasty_names_l_english.yml [hunk 3]
```diff
@@ -1931,7 +1931,6 @@
  dynn_Snowmantle:0 "Snowmantle"
  dynn_Redmane:0 "Redmane"
  dynn_Blackore:0 "Blackore"
- dynn_Blackforge:0 "Blackforge"
```
Comment (zumbak04) [inline localization/english/dynasties/wc_dynasty_names_l_english.yml:1934]: If you remove the localization line, you should leave a gap.

File: common/culture/cultures/elf.txt [hunk 4]
```diff
@@ -20,240 +20,114 @@
 		color = rgb { 255 255 0 }
```
Comment (zumbak04) [inline common/culture/cultures/elf.txt:20]: Error:
```
[20:38:29][culture_names.cpp:103]: Missing loc for name 'Tel'arn' in culture 'nightborne'
```

**Review by zumbak04 — CHANGES_REQUESTED** [review]

**Review by IronyOwl — APPROVED** [review]
High/Blood Elves look good, both dynasty names in common/dynasties and those not listed there were showing up.
Did get "Valandil" as a BE dynasty name, unsure if that's intended or a first name snuck in there.

Night Elf dynasty names look good, and had both historical and not.
Was getting a few weird first names, though. "Gedel" might be a NE name, but Luther, Hodle'mur, and Amywiar sound strange as male night elven first names.

Nightborne seemed okay, don't know enough about them to say much more. I wasn't able to get any historical dynastic names, but their dynastic namefile is small and doesn't list cultures so I assume that's intentional.

Satyr first names are very nicely thematic, I like those a lot. Dynastic names seemed good but I wasn't able to get any outside the dynasty file.
I did get one "dynn_Sargeron" as a dynasty name, however.

Naga first names were also very nice, and I got both historical and non from their dynasty names.
It seems like there's a bit of a gap between the historical and ahistorical ones, though. Almost all of the historical ones are two-part words (Hatecoil, Blackscale, Daggerspine), and almost all of the rest are slithery sounds separated by a ' (Naz'jar, Idra'kess, Vashj'elan). The ahistorical ones, by contrast, seem to be a mix of that latter category (Sral'ash, Shaz'jes, Grith'ja), and singular optionally slithery words (Sralar, Cithri).

**Comment by Loralius** [comment]
>High/Blood Elves look good, both dynasty names in common/dynasties and those not listed there were showing up.
Did get "Valandil" as a BE dynasty name, unsure if that's intended or a first name snuck in there.

Everything here is intended. Names are drawn from 3 files for different purposes. The culture .txt contains all the first and dynasty names used during character creation and NPC generation. The dynasty .txt provides fixed dynasties for the characters in listed in history, and the characters listed in history are assigned fixed first names. Almost all minor/ahistorical characters in the history files are using generated (old) names which I have replaced and I hope to replace them in a future branch, at which point the player should only ever see names from the lists in the culture and dynasty .txt.

Valandril is a surname from this [RPG source](https://wowpedia.fandom.com/wiki/Ashra_Valandril).

> Night Elf dynasty names look good, and had both historical and not.
Was getting a few weird first names, though. "Gedel" might be a NE name, but Luther, Hodle'mur, and Amywiar sound strange as male night elven first names.

These are residual names from the old lists (currently in use on master). The Thalassian bias at Blizzard HQ is laid bare as that culture has as many named NPCs as the rest of the Elven group combined. Roughly a third of the name additions are my own  "lore-friendly" names in addition to named lore character names for most of the subcultures and still, they are not enough to completely throw out all of the old names (repetition is an issue) except for that of the Thalassian group. I intend to come back and add more "lore-friendly" names from time to time but for now I need to move onto the other culture groups. If anyone wishes to add their own suggestions, check the `elf.txt` in cultures and leave suggestions. 

> Nightborne

They are severely lacking in named lore characters so the ones I did add are probably being drowned out by the rest of the generated names. Apart from brainstorming more lore-inspired names, there is potential for some (but not too many) dynastic names from the other subcultures to be shared with the Nightborne, esp. Highborne and Thalassian, to reflect their shared heritage. Because they are not currently implemented, I expect they will receive a targeted cultural iteration once we introduce the Broken Isles.

> I did get one "dynn_Sargeron" as a dynasty name, however.

Great catch. I will fix this. Much appreciated :)

On Satyr and Naga, they are both in the unfortunate categories of being common hostile mobs in WoW and thus named characters are sparse. They will each require a more focused PR in the future.

**Review by zumbak04 — APPROVED** [review]
LGTM!
The code looks clean, didn't find any errors.

**Pull request merged**